File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res=floor(nums.size()/2);
        map<int,int>m;
        for(auto it:nums)m[it]++;
        
        for(auto it:m){
            if(it.second>res) return it.first;
        }
        
        return -1;
    }
        
};

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int candidate=-1,count=0;
        
        for(int i=0;i<nums.length;i++){
            if(count==0){
                candidate=nums[i];
            }
            if(nums[i]==candidate) count++;
            else count--;
        }
        
        return candidate;
    }
}